predict.py

```diff
@@ -0,0 +1,105 @@
+import torch
+import click
+from utils.model_utils import AVAILABLE_MODELS, get_model_info, get_model_info_by_attributes, get_model_path
+from models.model_factory import get_model_class
+import logging
+from dotenv import load_dotenv
+import os
+import cv2
+import numpy as np
+
+load_dotenv()
+DATA_FOLDER_PATH = os.getenv("DATA_FOLDER_PATH")
+
+logging.basicConfig() 
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+@click.command()
+@click.option('-i', '--input', required=True, type=str, help="Path to input image.")
+@click.option('-id', '--identifier', type=str, help="Model id. You can print model info with help.py.")
+@click.option('-m', '--model', type=click.Choice(AVAILABLE_MODELS, case_sensitive=False), help='Model architechture.')
+@click.option('-n', '--num-classes', type=int, help='Number of classes (2 in binary case, 4 in multi-class case).')
+@click.option('-d', '--dataset', type=str, help='Name of the dataset model is trained on.')
+@click.option('-v', '--verbose', is_flag=True, show_default=True, default=False, help='Print verbose logs.')
+def predict(input, identifier, model, num_classes, dataset, verbose):
+
+  if verbose:
+    logger.setLevel(logging.DEBUG)
+
+  if not identifier and (not model or not num_classes or not dataset):
+    raise ValueError("You must provide either model id or model architechture and num of classes and dataset.")
```

Comment:
Based on the error message, should the ands and ors be the other way around in the conditional?
```suggestion
  if not identifier or (not model and not num_classes and not dataset):
    raise ValueError("You must provide either model id or model architechture and num of classes and dataset.")
```